NYCPlanning/db-zap-opendata
# refresh repo

## changes
* add a dev container
* replace use of `poetry` with `pip-tools` and requirements.txt (as we've been doing in other repos)
* use latest versions of python packages
* make necessary changes to python code to use latest packages
* add type hints to python functions
* revise some print statements, error handling, and github action names for clarity
* replace use of deprecated `set-output` with `echo "output_name=::output_value" >> "$GITHUB_OUTPUT"` (see [github notice](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/))
* update versions of actions used on our github action workflows (`actions/checkout`, `actions/setup-python`)
* add github action for PR testing and minimal python test
* add placeholder github action for TODO addition of MapZAP data (#38) (a new github action can't be called manually until it's been merged)

## tests
* [ran](https://github.com/NYCPlanning/db-zap-opendata/actions/runs/4968511375) "Export Single Open Dataset from CRM to DO" action on this branch
* added minimal PR tests which are passing
* note: have not tested "Weekly Export" action with this branch. it takes ~1.5 hours and I didn't want to risk conflicting with [the scheduled Sunday run on main](https://github.com/NYCPlanning/db-zap-opendata/actions/runs/4975096160)
